src/auth/use-authentication.ts

```diff
@@ -0,0 +1,31 @@
+import { AuthorizationGuard, AuthorizationPayload, instanceToScope, Thread } from '@serlo/authorization'
+import { gql } from 'graphql-request'
+
+import { useGraphqlSwrWithAuth } from '@/api/use-graphql-swr'
+import { useInstanceData } from '@/contexts/instance-context'
+import { Instance } from '@serlo/api'
+
+export function useAuthentication() {
+  // TODO: this should also work if authorization === null
+  const instance = useInstanceData()
+  const scope = instanceToScope(instance.lang as Instance)
+  const { data } = useGraphqlSwrWithAuth<{
```

Comment:
TODO: `useGraphqlSwrWithAuth` throws an error if user is not authenticated. Should probably add an option like `requireAuth` that falls back to an request without token.